File: rules/csharp/tests/cs.sql-injection/Sql.cs
class SqlTests {
  void Direct() {
    var id = Request.QueryString["id"];
    var cmd = new SqlCommand("SELECT * FROM u WHERE id=" + id); // crit:expect cs.sql-injection
  }
  void CommandText(SqlCommand c) {
    var id = Request.Form["id"];
    c.CommandText = "SELECT * FROM u WHERE id=" + id; // crit:expect cs.sql-injection
  }
  void Constant() {
    var cmd = new SqlCommand("SELECT * FROM u WHERE active=1"); // crit:expect-not cs.sql-injection
  }
  void Internal(string id) {
    var cmd = new SqlCommand("SELECT * FROM u WHERE id=" + id); // crit:expect-not cs.sql-injection
  }
}

File: rules/go/tests/go.sql-injection/sql.go
package main

import "net/http"

func direct(r *http.Request, db DB) {
	db.Query("SELECT * FROM u WHERE id = " + r.FormValue("id")) // crit:expect go.sql-injection
}

func viaVar(r *http.Request, db DB) {
	id := r.URL.Query().Get("id")
	// crit:expect go.sql-injection
	db.Query("SELECT * FROM u WHERE id = " + id)
}

func constant(db DB) {
	db.Query("SELECT * FROM u WHERE active = 1") // crit:expect-not go.sql-injection
}

func internal(db DB, id string) {
	db.Query("SELECT * FROM u WHERE id = " + id) // crit:expect-not go.sql-injection
}
